tests/libvmi/factory.go

```diff
@@ -29,42 +29,44 @@ import (
 const (
 	DefaultTestGracePeriod int64 = 0
 	DefaultVmiName               = "testvmi"
+	userDataSetPassword          = `#!/bin/bash
+	echo "fedora" | passwd fedora --stdin
+	echo `
 )
 
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedora, opts...)
+	return newFedora(cd.ContainerDiskFedora, userDataSetPassword, opts...)
 }
 
 // NewTestToolingFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 // This image has tooling for the guest agent, stress, and more
 func NewTestToolingFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraTestTooling, opts...)
+	return newFedora(cd.ContainerDiskFedoraTestTooling, userDataSetPassword, opts...)
 }
 
 // NewSriovFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options, the
 // image used include Guest Agent and some moduled needed by SRIOV.
 func NewSriovFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraSRIOVLane, opts...)
+	userData := `#!/bin/bash
+	echo `
```

Comment:
why does userdata need to be defined here if the script doesn't do anything?

Reply:
It doesn't need to be, it just so it will be easy to add it, i can remove it (verified it works in all options)

Reply:
Basically it isn't needed, 
but from what i remember tests which use NetworkData do need this, else they won't run Cloud-init at all, 
so felt its better to keep it, i can verify it if needed.
Thought its safe to keep it and it will also allow easier modification of the UserData